graphql/util.go

```diff
@@ -10,10 +10,18 @@ package graphql
 //	type T struct { E; NoUnmarshalJSON }
 // where E has an UnmarshalJSON method, T will not inherit it, per the Go
 // selector rules: https://golang.org/ref/spec#Selectors.
+//
+// It also works for marshaling, in the same way (despite the name).
```

Comment:
What do you think about naming this NoMarshalJSON? It seems a little more expected that a type with this name would also handle unmarshaling vs. the other way around. (Another option is to have a completely separate type so marshalnig and unmarshaling can be handled separately.)

Reply:
I'm not super excited to rename it, but maybe I'll just add a separate NoMarshalJSON; we only ever need one.